server/src/main/java/io/crate/planner/operators/EquiJoinDetector.java

```diff
@@ -76,54 +76,46 @@ private static class Visitor extends SymbolVisitor<Context, Void> {
         public Void visitFunction(Function function, Context context) {
             String functionName = function.name();
             switch (functionName) {
-                case AndOperator.NAME:
-                    for (Symbol arg : function.arguments()) {
-                        arg.accept(this, context);
+                case NotPredicate.NAME -> {
+                }
+                case OrOperator.NAME -> {
+                    context.isHashJoinPossible = false;
```

Comment:
Shouldn't we exit here already? Otherwise if we have `t1.a = t1.b OR t1.c = t2.d`, it will be detected as hashjoin?

Reply:
No, it evaluates to false.